SpatialGDK/Source/SpatialGDK/Public/EngineClasses/SpatialShadowActor.h

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#pragma once
+#include "SpatialConstants.h"
+#include "SpatialShadowActor.generated.h"
+
+DECLARE_LOG_CATEGORY_EXTERN(LogSpatialShadowActor, Log, All);
+
+UCLASS(Transient)
+class SPATIALGDK_API USpatialShadowActor : public UObject
+{
+	GENERATED_BODY()
+
+public:
+	USpatialShadowActor(const FObjectInitializer& ObjectInitializer = FObjectInitializer::Get());
+
+	void Init(const Worker_EntityId InEntityId, AActor* InActor);
+
+	void CheckUnauthorisedDataChanges(const Worker_EntityId InEntityId, AActor* InActor);
+
+	void Update(const Worker_EntityId InEntityId, AActor* InActor);
+
+private:
+	Worker_EntityId EntityId;
+	AActor* Actor;
+	FString ReplicatedPropertyHash;
+	FString CreateHash(AActor* InActor);
+
+	// Store a list of actors class names that currently violate the non-auth changes so that the user is not spammed.
+	// TODO: link PR to investigate these cases
+	UPROPERTY()
+	TArray<FString> SuppressedActors;
```

Comment:
This should be a constant array somewhere, instead of being associated to every single instance of the shadow data. And this should also be something faster to check, like a UClass*

Reply:
+1 - I missed it was a string.

Reply:
Good point. I've made this a constant array in the new utility class. I changed from FString to FName for performance but if we use UClass we will be restricted to only add classes from the Engine/ GDK and we will not be able to add Test classes if needed.
I've also added a similar mechanism to suppress properties as I noticed many duplicate property non-auth changes e.g. RemoteRole and Role. So it makes sense for these to be consistent.